include/effects/matrix/PatternCyclicCA.h

```diff
@@ -0,0 +1,581 @@
+
+        // Accumulate time for frame and pattern timers
+        m_frameTimer += delta;
+        m_patternTimer += delta;
+
+        // Check if the pattern has died (no state changes in the last generation)
+        // or if the pattern lifetime has been reached (if lifetime is not 0)
+        // Re-seed the pattern if either condition is met.
+        // Added check for m_lastDrawTime > 0 to prevent re-seeding immediately on start
+        if ((m_sum == 0 && m_lastDrawTime > 0) || (m_lifetime > 0 && (m_patternTimer >= m_lifetime)))
+        {
+            SeedCA();
+            m_patternTimer = 0; // Reset pattern timer
+            m_frameTimer = 0;   // Reset frame timer
+            // Note: m_sum is reset within SeedCA
+        }
+
+        // Check if enough time has passed to calculate the next generation.
+        if (m_frameTimer >= m_speed)
+        {
+            // Calculate the next generation using the currently selected method (GH or CCA).
+            (this->*m_calcNextGen)();
+            m_frameTimer = 0; // Reset frame timer
+            // Note: m_sum is updated within the generation functions.
+        }
+
+        // Render the current state of the CA grid to the matrix.
+        g()->fillScreen(BLACK16); // Clear the screen before drawing the new frame
+
+        for (int y = 0; y < MATRIX_HEIGHT; y++)
+        {
+            for (int x = 0; x < MATRIX_WIDTH; x++)
+            {
+                // Get the cell state from the current buffer (m_pb2).
+                uint8_t cellState = (*m_pb2)[x][y];
+                // Map the state to a color.
+                CRGB pixelColor = ColorFromCellState(cellState);
+                // Draw the pixel on the matrix.
+                g()->setPixel(x, y, pixelColor);
+            }
+        }
+    }
+
+    // Specify the desired base frame rate (actual rate controlled by 'speed')
+    virtual size_t DesiredFramesPerSecond() const override
+    {
+        return 60; // A reasonable base rate for the effect runner
+    }
+
+    // Indicate that this effect manages its own buffers
+    bool RequiresDoubleBuffering() const override
+    {
+        return false;
+    }
+
+    // Serialize the entire effect state to JSON (for saving/loading effect instances).
+    bool SerializeToJSON(JsonObject& jsonObject) override
+    {
+        // Serialize base class data first.
+        auto jsonDoc = CreateJsonDocument();
+        JsonObject root = jsonDoc.to<JsonObject>();
+        LEDStripEffect::SerializeToJSON(root); // Base class method
+
+        // Serialize our specific settings (using short keys for efficiency in JSON)
+        jsonDoc["ns"] = m_numStates;
+        jsonDoc["sp"] = m_speed;
+        jsonDoc["lt"] = m_lifetime;
+        jsonDoc["ex"] = m_excited;
+        jsonDoc["rf"] = m_refractory;
+        jsonDoc["th"] = m_threshold;
+        // Serialize the enum as a boolean for compatibility with the existing JSON structure
+        jsonDoc["mo"] = (m_mode == CAMode::Cyclic);
```

Comment:
- As mentioned before, these should be _three-letter_ acronyms.
- In all cases they **must** match the property names referred to in the deserialization constructor, because otherwise deserialization will always fail.